src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -287,6 +287,7 @@ class device {
             if (computeUnits.at( i ).m_type != IP_KERNEL)
                 continue;
 	    if (!skip_cu) {
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
We will need to push/pop the current state of GCC diagnostic while it is compiling a file.

```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
xclRead(m_handle, XCL_ADDR_KERNEL_CTRL, computeUnits.at( i ).m_base_address, &statusBuf, 4);
#pragma GCC diagnostic pop
```

Reply:
Added #pragma push/pop. Please review.